src/Hooks.php

```diff
@@ -83,6 +83,11 @@ public function dispatch($hook, $parameters = array()) {
 			return false;
 		}
 
+		if (!empty($parameters)) {
+			// Strip potential keys from the array to prevent them being interpreted as parameter names in PHP 8.0.
+			$parameters = array_values($parameters);
+		}
+
 		foreach ($this->hooks[$hook] as $priority => $hooked) {
 			foreach ($hooked as $callback) {
 				call_user_func_array($callback, $parameters);
```

Comment:
Why are we not using variadic arguments here instead?
```suggestion
				$callback(...$parameters);
```

Reply:
> Why are we not using variadic arguments here instead?

Historically because it wasn't possible previously as the minimum PHP version did not allow for the spread operator ;-)

As for changing it now... I'm a bit wary of doing so as the behaviour will change with regards to references and the method is called quite often with variables being passed by reference.

Also see https://core.trac.wordpress.org/ticket/47678#comment:31 - section 3.

Reply:
Then again... `array_values()` may already break the references...

Reply:
Yes, that's what I thought too. If we are using `array_values()`, it will be pretty much the same thing as using the variadic arguments operator.

Reply:
Well, I'm not 100%, I think this needs to be tested.

Reply:
I've tested this and added a unit test to safeguard this as well.

The test passed prior to the code changes and still passes with both the code changes in place, so I think we're good.

Reply:
(to my surprise, I may add)